osbenchmark/builder/installers/plugin_installer.py

```diff
@@ -0,0 +1,50 @@
+import logging
+import os
+
+from osbenchmark.builder.installers.installer import Installer
+from osbenchmark.builder.provision_config import BootstrapHookHandler
+from osbenchmark.builder.utils.config_applier import ConfigApplier
+from osbenchmark.builder.utils.path_manager import PathManager
+from osbenchmark.builder.utils.template_renderer import TemplateRenderer
+
+
+class PluginInstaller(Installer):
+    def __init__(self, plugin, executor, hook_handler_class=BootstrapHookHandler):
+        super().__init__(executor)
+        self.logger = logging.getLogger(__name__)
+        self.plugin = plugin
+        self.hook_handler = hook_handler_class(self.plugin)
+        if self.hook_handler.can_load():
+            self.hook_handler.load()
```

Comment:
Does this need to happen during the object construction itself? Would it make sense to move this to the install logic?

I'm not fundamentally against it but prefer lightweight constructors. Think of scenarios where these fail and what stack trace they leave - I feel it makes more sense if we say the plugin failed to install /load. Instead if the constructor throws an exception, I suspect there's things like class not found exception or something which isn't a good description of what happened.

Reply:
It could go in the `install` logic. The rationale for having it here is to fail fast if an exception is thrown when loading the `hook_handler`.

If this logic is moved to the `install` method, then the exception it creates on failure would only be thrown after the infrastructure has been provisioned and the necessary data downloaded and installed on the host. By failing in the constructor, no resources are created or work done for a scenario that couldn't possibly succeed.

Perhaps wrapping the exception with some documentation on the failure would be a solution to the vague description. Given that this `hook_handler` logic is used in multiple Installers, adding that handling into the factory generating Installer instances would be a good approach.